### ISSUE TITLE
`SurrogateTest` with `CMIShannon()` and `KSG1()` estimator doesn't produce correct results

### ISSUE THREAD

**Issue by kahaaga** [JuliaDynamics/CausalityTools.jl#210]
@bhannis Here's a MWE for the issue CMI issue you discovered

```julia
julia> using CausalityTools
julia> x, z = rand(100), rand(100);
julia> test_mi = SurrogateTest(MIShannon(), KSG1());
julia> test_cmi = SurrogateTest(CMIShannon(), KSG1());
julia> independence(test_mi, x, x)
`SurrogateTest` independence test result
-----------------------------------------------------------------------------------
H₀: "The first two variables are independent (given the 3rd variable, if relevant)"
Hₐ: "The first two variables are dependent (given the 3rd variable, if relevant)"
-----------------------------------------------------------------------------------
Estimated: 2.2023902679264586
Ensemble quantiles (100 permutations):
  (99.9%): 0.10273316693861609
  (99%):   0.09602323042253126
  (95%):   0.005943171154352587
p-value:   0.0
  α = 0.05:  ✓ Evidence favors dependence
  α = 0.01:  ✓ Evidence favors dependence
  α = 0.001: ✓ Evidence favors dependence
  
  julia> independence(test_cmi, x, x, z)
`SurrogateTest` independence test result
-----------------------------------------------------------------------------------
H₀: "The first two variables are independent (given the 3rd variable, if relevant)"
Hₐ: "The first two variables are dependent (given the 3rd variable, if relevant)"
-----------------------------------------------------------------------------------
Estimated: -1.2147778711621595
Ensemble quantiles (100 permutations):
  (99.9%): 0.22498610454627826
  (99%):   0.1282710018619384
  (95%):   0.07460434232826373
p-value:   1.0
  α = 0.05:  ✖ Independence cannot be rejected
  α = 0.01:  ✖ Independence cannot be rejected
  α = 0.001: ✖ Independence cannot be rejected
```

The last test should yield precisely the same answer as the first, because `z` is independent of `x`.

**Comment by kahaaga** [comment]
I presume this is due to an error in `estimate(::ConditionalMutualInformation, est::MutualInformationEstimator, x, y, z)`, but will need to check

**Comment by kahaaga** [comment]
Fixed in #214
